packages/tests/src/pom/form.ts

```diff
@@ -0,0 +1,56 @@
+import { Locator, type Page } from "@playwright/test";
+import type { FactorMethod } from "@slashid/slashid";
+
+type FactorMethodLabel = "OTP via email" | "Email link";
+
+function getLabelFromFactorMethod(method: FactorMethod): FactorMethodLabel {
+  // @ts-expect-error add missing labels when needed
+  const methodToLabel: Record<FactorMethod, FactorMethodLabel> = {
+    otp_via_email: "OTP via email",
+    email_link: "Email link",
+  };
+
+  return methodToLabel[method];
+}
+
+export class FormPage {
```

Comment:
Page object model for the `/form` page. High level idea:
- test scenarios should not depend on locators as they are low level
- instead test scenarios depend on page object models
- a page object model hides away the locators and exposes a high level API
- test scenario can then be written using a more natural language based on the page object model

Reply:
At the same time this makes the test scenarios more readable and also easier to maintain, given that multiple scenarios can target the same page. This way they will reuse the same selectors instead of duplicating them.